internal/engine/buildcontroller_test.go

```diff
@@ -543,6 +543,37 @@ func TestBuildControllerManualTriggerWithoutFileChangesForceUpdates(t *testing.T
 	})
 }
 
+func TestBuildControllerManualTriggerWithFileChangesSinceLastSuccessfulBuildButBeforeLastBuild(t *testing.T) {
+	f := newTestFixture(t)
+	defer f.TearDown()
+	mName := model.ManifestName("foobar")
+
+	manifest := f.newManifest(mName.String())
+	manifests := []model.Manifest{manifest}
+	f.Start(manifests, true)
+
+	f.nextCallComplete()
+
+	f.b.nextBuildFailure = errors.New("build failure!")
+	f.fsWatcher.events <- watch.NewFileEvent(f.JoinPath("main.go"))
+	f.nextCallComplete()
+
+	f.store.Dispatch(server.AppendToTriggerQueueAction{Name: mName})
+	call := f.nextCallComplete()
+	state := call.oneState()
+	assert.Equal(t, []string{f.JoinPath("main.go")}, state.FilesChanged())
+	assert.True(t, state.NeedsForceUpdate)
```

Comment:
might put a comment to emphasize that this is the behavior you're specifically trying to test... (OR make the test name longer to say that you expect a force udpate?) but lgtm!